src/main/kotlin/blackjack/domain/player/Dealer.kt

```diff
@@ -2,36 +2,232 @@ package blackjack.domain.player
 
 import blackjack.domain.card.Card
 import blackjack.domain.card.Cards
-import blackjack.domain.game.Rule
+import blackjack.domain.game.Bet
+import blackjack.domain.game.Credit
+import blackjack.domain.game.GameResult
 import blackjack.domain.game.Score
+import blackjack.domain.game.ScoreResult
 
-class Dealer(profile: Profile, cards: Cards = Cards.EMPTY) : BlackJackPlayer(profile, cards) {
+class Dealer(profile: Profile, cards: Cards = Cards.EMPTY, credit: Credit = Credit.from(0)) :
+    BlackJackPlayer(profile, cards, credit) {
 
     override fun receiveCard(card: Card): Player {
         if (!canReceiveCard()) {
             return this
         }
-        return Dealer(profile, cards.addCards(card))
+        return Dealer(profile, cards.addCards(card), credit)
     }
 
     override fun turnOff(): Player {
-        return Dealer(profile.turnOff(), cards)
+        return Dealer(profile.turnOff(), cards, credit)
     }
 
     override fun turnOn(): Player {
-        return Dealer(profile.turnOn(), cards)
+        return Dealer(profile.turnOn(), cards, credit)
     }
 
     override fun canReceiveCard(): Boolean {
         return getHighestPoint() <= CAN_ACHIEVE_POINT
     }
 
-    override fun judgeResult(gamer: List<Player>, rule: Rule): Map<Player, List<Score>> {
-        return rule.judge(this, gamer)
+    override fun holdBetting(getPlayerBetting: Bet): Player {
+        return Dealer(profile, cards)
+    }
+
+    override fun winBetting(player: Player): Player {
+        return Dealer(profile, cards, credit.receiveCredit(player.getPlayerCredit()))
+    }
+
+    override fun winBetting(): Player {
+        return Dealer(profile, cards, credit)
+    }
+
+    override fun loseBetting(): Player {
+        return Dealer(profile, cards, credit.minus(credit))
+    }
+
+    override fun loseBetting(player: Player): Player {
+        return Dealer(profile, cards, credit.minus(player.getPlayerCredit()))
+    }
+
+    override fun winBlackJack(): Player {
+        return Dealer(profile, cards, credit.receiveBlackJackCredit())
+    }
+
+    override fun loseBlackJack(player: Player): Player {
+        return Dealer(profile, cards, credit.subtractBlackJack(player.getPlayerCredit()))
+    }
+
+    override fun minusCredit(credit: Credit): Player {
+        return Dealer(profile, cards, credit.minus(credit))
```

Comment:
```suggestion
       return Dealer(profile, cards, this.credit-credit)
```

연산자 오버로딩을 적용하시면 좋을거같아요.